system-tests/tests/smoke/cre/por_test.go

```diff
@@ -675,6 +825,69 @@ func TestCRE_OCR3_PoR_Workflow_SingleDon_MultipleWriters_MockedPrice(t *testing.
 	waitForFeedUpdate(t, testLogger, priceProvider, setupOutput, 5*time.Minute)
 }
 
+// Verifies that workflow can read a given contract and write it to a feed.  Feed is read by fetching the latest bundle.
+// Config file to use: environment-one-don-read-contract.toml
+func TestCRE_OCR3_ReadBalance_Workflow_SingleDon_MockedPrice(t *testing.T) {
+	testLogger := framework.L
+	expectedReadAmount := big.NewInt(99)
+
+	// Load and validate test configuration
+	in, err := framework.Load[TestConfig](t)
+	require.NoError(t, err, "couldn't load test config")
+	validateEnvVars(t, in)
+	require.Len(t, in.NodeSets, 1, "expected 1 node set in the test config")
+
+	// Assign all capabilities to the single node set
+	mustSetCapabilitiesFn := func(input []*ns.Input) []*keystonetypes.CapabilitiesAwareNodeSet {
+		return []*keystonetypes.CapabilitiesAwareNodeSet{
+			{
+				Input:              input[0],
+				Capabilities:       append(SinglePoRDonCapabilitiesFlags, "read-contract"),
+				DONTypes:           []string{keystonetypes.WorkflowDON, keystonetypes.GatewayDON},
+				BootstrapNodeIndex: 0, // not required, but set to make the configuration explicit
+				GatewayNodeIndex:   0, // not required, but set to make the configuration explicit
+			},
+		}
+	}
+
+	// fake price provider without a data provider, price will be read on chain
+	priceProvider := &FakePriceProvider{
+		testLogger: testLogger,
+		expectedPrices: map[string][]*big.Int{
+			cleanFeedID(in.WorkflowConfigs[0].FeedID): {
+				expectedReadAmount,
+			},
+		},
+		actualPrices: make(map[string][]*big.Int),
+	}
```

Comment:
I don't think you really need to use it. It doesn't really fit into the readcontract test and I don't think we should use it only to satisfy the interface.

I'd either just use a simple comparison of the value you fund the contract with, with what is send by the workflow to df cache. Or read from the contract directly and compare it with value in DF cache. No need to involve price provider.

Reply:
the price provider is storing the value read from the DF cache.